src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php

```diff
@@ -343,6 +343,7 @@ public function getConfig(): ?array
         return [
             'teaser' => $this->teaserProviderPool->getConfiguration(),
             'versioning' => $this->versioningEnabled,
+            'webspaces' => $this->webspaceManager->getWebspaceCollection()->getWebspaces(),
```

Comment:
Here I am just outputting all the webspaces. The `WebspaceController` had a separate flag `checkForPermissions`, based on which was decided if only the webspaces for which the user has the `VIEW` permission or all of them should be returned.

Wondering how we should implement that now... Maybe add a `_permissions` object, similar to what we already have in the responses of the `PageController`.

Then the `webspaceStore` could offer two methods:
- `getAllWebspaces()` returning all the webspaces in the system
- `getGrantedWebspaces()` only returning webspaces for which the current user has the `VIEW` permission

@sulu/core-team WDYT?

Reply:
The `_permissions` sounds good for me. With `getGrantedWebspaces` I'm not sure, as I think it mostly is forgotten to use that method so I would say that `getAllWebspaces` should by default return only the webspaces I have permissions for like the api did and add a optional bool flag or additional method to get also the ones without I have permissions for.

Reply:
Actually my reasoning was that there is no `getWebspaces` method, so that it is clear if I get all of them, or just the ones I was granted for. I thought that should be pretty clear this way. Boolean flags in functions tend to explode, therefore I would like to have to separate methods for that (actually I am currently even favouring computed properties, but my previous statement is still the same).

Reply:
@danrot sounds good for me 👍